1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,5 +1,61 @@
-function setAlarm() {}
+function setAlarm() {
+  const input = document.getElementById("alarmSet").value;
+  const setBtn = document.getElementById("set");
+  const stopBtn = document.getElementById("stop");
+  const timer = document.getElementById("timeRemaining");
 
+  // Make Set Alarm Button working:
+  setBtn.addEventListener("click", () => {
+    // check if the input is a positive integer.
+    input <= 0 || input === ""
+      ? alert("please insert a positive integer")
+      : alert("The count down will start now");
+
+    // saving input value ,before clearing it's value.
+    let time = input;
+    // clearing the input value.
+    document.getElementById("alarmSet").value = "";
+
+    // making
+    let timerInterval = setInterval(CountDown, 1000);
+
+    function CountDown() {
+      let min;
+      let sec;
+      if (time >= 0) {
+        // calculate minutes and seconds with input value.
+        min = Math.floor(time / 60);
+        min < 10 ? (min = `0${min}`) : min;
+        sec = time - min * 60;
```

Comment:
You need to be really careful here mixing up your data types. You started with `min` as a number and then conditionally reassigned it to a string value on line 28. Here you're trying to do some maths with a variable which might be numeric and might be text. Javascript will kind of "just work" a lot of the time, and other times it'll get it horribly wrong and you'll have all kinds of weird bugs to deal with.
Be really strict with yourself only using variables for one kind of data.

Reply:
thanks Jon I will keep it mind and try to be more careful about data types